pkg/arvo/sys/hoon.hoon

```diff
@@ -10194,6 +10201,119 @@
       {* * ~}   [dov $(dom l.dom)]
       {* * *}   [dov $(dom l.dom) $(dom r.dom)]
     ==
+    ::
+    ::  all the below arms are used for gol checking and should have no
+    ::  effect other than giving more specific errors
+    ::
+    ::  all the possible types we could be expecting.
+    ::
+    +$  gol-type
+      $~  %noun
+      $@  %noun
+      $%  [%cell p=type q=type]
+          [%core p=type q=coil]
+          [%fork p=(set gol-type)]
+      ==
```

Comment:
Are we sure that there's never any need to declare molds in separate cores/chapters from their use-sites? I *think* the seminoun system made that pattern unnecessary, but I haven't verified that.

Reply:
I don't understand, what about this affects that?

Reply:
You've defined `$gol-type` in the same core (and chapter) as the arms that use it. That used to be frowned upon. As far as I know it's not a problem anymore (but when it was, I didn't understand it well) ...